workers.py

```diff
@@ -26,6 +27,26 @@ def parse(_):
 MAX_RETRIES = 3
 
 
+def add_metrics(method: str, result: bool):
+    """Add Prometheus Counter Metrics for Request types GET and POST.
+
+    :param method: "get", "post" etc.
+    :param result: True/False based on the Request response
+    """
+    if method == 'get' and result is True:
```

Comment:
spaghetti means intertwined code. That particular code is indecisive, yes - but not intertwined :)
The reason for using so many conditionals was really because the way you have written the `_retryable` method - it's *way* too generic and does not really help with this task of adding metric checkpoints.

Anyway, like I mentioned above, that function is now gone